web_poet/example.py

```diff
@@ -0,0 +1,86 @@
+from asyncio import run
+from typing import Any, Dict, Optional, Type
+
+import andi
+from requests import get
+from url_matcher import URLMatcher
+
+from . import default_registry
+from .page_inputs import HttpClient, HttpResponse, PageParams
+from .pages import ItemPage, is_injectable
+
+
+class _HttpClient:
+    async def get(self, url: str) -> HttpResponse:
+        return _get_http_response(url)
+
+
+def _get_page_class(url: str, item_class: Type) -> Type[ItemPage]:
+    url_matcher = URLMatcher(
+        {
+            rule.use: rule.for_patterns
+            for rule in default_registry.get_rules()
+            if (rule.to_return is None or rule.to_return == item_class)
```

Comment:
why do you need to consider classes with to_return==None?

Reply:
I assumed, if no `to_return` class is specified, that they are valid options whatever the value of  `to_return` is. But I guess it makes sense to consider that, by passing a specific item class to `get_item`, only page object classes which report returning such an item class should be elegible.

Reply:
> I assumed, if no to_return class is specified, that they are valid options whatever the value of to_return is. 

Let's say you request MyCustomProduct and MyCustomproductList items on example.com, and there is some a `handle_urls("example.com")` page object. Most likely, it knows nothing about either of the MyCustom items. And you can't possibly use it to return both items. So, if to_return is not specified, it means we shouldn't use this page object as a way to return an item.